src/plugins/discover/public/application/main/services/discover_internal_state_container.ts

```diff
@@ -12,13 +12,15 @@ import {
   ReduxLikeStateContainer,
 } from '@kbn/kibana-utils-plugin/common';
 import { DataView, DataViewListItem } from '@kbn/data-views-plugin/common';
+import { Filter } from '@kbn/es-query';
 import { DataTableRecord } from '../../../types';
 
 export interface InternalState {
   dataView: DataView | undefined;
   savedDataViews: DataViewListItem[];
   adHocDataViews: DataView[];
   expandedDoc: DataTableRecord | undefined;
+  datasetFilters: Filter[];
```

Comment:
I think it would be best to give this a more generic name since consumers other than Logs Explorer may also choose to use this in the future. I'm not really sure of a good generic name for it though, maybe something like `implicitFilters`? I'm thinking something that communicates that these filters are set programatically and not tied to the default Discover UI.

Reply:
`customFilters` is a good one, I'd be happy if we went with that.